couchdbkit/schema/base.py

```diff
@@ -152,14 +152,32 @@ def save_docs(cls, docs, use_uuids=True, all_or_nothing=False):
 
         """
         db = cls.get_db()
-        docs_to_save= [doc for doc in docs if doc._doc_type == cls._doc_type]
-        if not len(docs_to_save) == len(docs):
+        if any(doc._doc_type != cls._doc_type for doc in docs):
             raise ValueError("one of your documents does not have the correct type")
-        db.bulk_save(docs_to_save, use_uuids=use_uuids, all_or_nothing=all_or_nothing)
+        db.bulk_save(docs, use_uuids=use_uuids, all_or_nothing=all_or_nothing)
 
     bulk_save = save_docs
 
     @classmethod
+    def delete_docs(cls, docs, all_or_nothing=False, empty_on_delete=False):
+        """ Bulk delete documents in a database
+
+        @params docs: list of couchdbkit.schema.Document instance
+        @param all_or_nothing: In the case of a power failure, when the database
+        restarts either all the changes will have been saved or none of them.
+        However, it does not do conflict checking, so the documents will
+        be committed even if this creates conflicts.
+        @param empty_on_delete: default is False if you want to make
+        """
+        db = cls.get_db()
+        if any(doc._doc_type != cls._doc_type for doc in docs):
+            raise ValueError("one of your documents does not have the correct type")
+        db.bulk_delete(docs, all_or_nothing=all_or_nothing,
```

Comment:
Yup, it works - all of these bulk update methods call `db.save_docs` under the hood, which calls [`_maybe_serialize`](https://github.com/dimagi/couchdbkit/blob/master/couchdbkit/client.py#L50-L62) on each doc eventually.  `Document.save_docs` (which this is modeled after) actually *requires* that you pass in Documents so it can verify that you are only passing docs of the appropriate type.